test/e2e_node/image_pull_test.go

```diff
@@ -0,0 +1,303 @@
+						if event.InvolvedObject.Name == testpod.Name {
+							pulled, err := getDurationsFromPulledEventMsg(event.Message)
+							imagePulled[testpod.Name] = pulled
+							framework.ExpectNoError(err)
+							break
+						}
+					}
+				} else if event.Reason == kubeletevents.PullingImage {
+					podStartTime[event.InvolvedObject.Name] = event.CreationTimestamp
+				}
+			}
+			gomega.Expect(len(testpods)).To(gomega.BeComparableTo(len(imagePulled)))
+
+			// skip if pod1 pulling time and pod2 pulling time are not overlapped
+			if podStartTime[testpods[0].Name].Time.Before(podStartTime[testpods[1].Name].Time) {
+				if podEndTime[testpods[0].Name].Time.Before(podStartTime[testpods[1].Name].Time) {
+					e2eskipper.Skipf("pod1 pulling time and pod2 pulling time are not overlapped")
+				}
+			} else {
+				if podEndTime[testpods[1].Name].Time.Before(podStartTime[testpods[0].Name].Time) {
+					e2eskipper.Skipf("pod1 pulling time and pod2 pulling time are not overlapped")
+				}
+			}
+
+			// as this is serialize image pulling, the waiting duration should be almost double the duration with the pulled duration.
+			// use 1.5 as a common ratio to avoid some overlap during pod creation
+			if float32(imagePulled[testpods[1].Name].pulledIncludeWaitingDuration/time.Millisecond)/float32(imagePulled[testpods[1].Name].pulledDuration/time.Millisecond) < 1.5 &&
+				float32(imagePulled[testpods[0].Name].pulledIncludeWaitingDuration/time.Millisecond)/float32(imagePulled[testpods[0].Name].pulledDuration/time.Millisecond) < 1.5 {
+				framework.Failf("At least, one of the pull duration including waiting %v/%v should be similar with the pulled duration %v/%v",
+					imagePulled[testpods[1].Name].pulledIncludeWaitingDuration, imagePulled[testpods[0].Name].pulledIncludeWaitingDuration, imagePulled[testpods[1].Name].pulledDuration, imagePulled[testpods[0].Name].pulledDuration)
+			}
+		})
+
+	})
+})
+
+func prepareAndCleanup(ctx context.Context, f *framework.Framework) (testpods []*v1.Pod) {
+	// cuda images are > 2Gi and it will reduce the flaky rate
+	image1 := imageutils.GetE2EImage(imageutils.Httpd)
+	image2 := imageutils.GetE2EImage(imageutils.HttpdNew)
+	node := getNodeName(ctx, f)
+
+	testpod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testpod",
+			Namespace: f.Namespace.Name,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{{
+				Name:            "testpod",
+				Image:           image1,
+				ImagePullPolicy: v1.PullAlways,
+			}},
+			NodeName:      node,
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+	testpod2 := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testpod2",
+			Namespace: f.Namespace.Name,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{{
+				Name:            "testpod2",
+				Image:           image2,
+				ImagePullPolicy: v1.PullAlways,
+			}},
+			NodeName:      node,
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+	testpods = []*v1.Pod{testpod, testpod2}
```

Comment:
Would 2 pods be enough for all future tests? Would it make sense to parametrize pod and image names?

Reply:
We may add more pods later.